core/src/main/java/com/google/cloud/sql/core/Connector.java

```diff
@@ -0,0 +1,136 @@
+  }
+
+  /** Extracts the Unix socket argument from specified properties object. If unset, returns null. */
+  private static String getUnixSocketArg(ConnectionConfig config) {
+    String unixSocketPath = config.getUnixSocketPath();
+    if (unixSocketPath != null) {
+      // Get the Unix socket file path from the properties object
+      return unixSocketPath;
+    } else if (System.getenv("CLOUD_SQL_FORCE_UNIX_SOCKET") != null) {
+      // If the deprecated env var is set, warn and use `/cloudsql/INSTANCE_CONNECTION_NAME`
+      // A socket factory is provided at this path for GAE, GCF, and Cloud Run
+      logger.warning(
+          String.format(
+              "\"CLOUD_SQL_FORCE_UNIX_SOCKET\" env var has been deprecated. Please use"
+                  + " '%s=\"/cloudsql/INSTANCE_CONNECTION_NAME\"' property in your JDBC url"
+                  + " instead.",
+              ConnectionConfig.UNIX_SOCKET_PROPERTY));
+      return "/cloudsql/" + config.getCloudSqlInstance();
+    }
+    return null; // if unset, default to null
+  }
+
+  Socket connect(ConnectionConfig config) throws IOException {
+    // Connect using the specified Unix socket
+    String unixSocket = getUnixSocketArg(config);
+    String unixPathSuffix = config.getUnixSocketPathSuffix();
+    if (unixSocket != null) {
+      // Verify it ends with the correct suffix
+      if (unixPathSuffix != null && !unixSocket.endsWith(unixPathSuffix)) {
+        unixSocket = unixSocket + unixPathSuffix;
+      }
+      logger.info(
+          String.format(
+              "Connecting to Cloud SQL instance [%s] via unix socket at %s.",
+              config.getCloudSqlInstance(), unixSocket));
+      UnixSocketAddress socketAddress = new UnixSocketAddress(new File(unixSocket));
+      return UnixSocketChannel.open(socketAddress).socket();
+    }
+
+    DefaultConnectionInfoCache instance = getConnection(config);
+    try {
+
+      SSLSocket socket = instance.createSslSocket(this.refreshTimeoutMs);
+
+      // TODO(kvg): Support all socket related options listed here:
+      // https://dev.mysql.com/doc/connector-j/en/connector-j-reference-configuration-properties.html
+      socket.setKeepAlive(true);
+      socket.setTcpNoDelay(true);
+
+      String instanceIp = instance.getPreferredIp(config.getIpTypes(), refreshTimeoutMs);
+
+      socket.connect(new InetSocketAddress(instanceIp, serverProxyPort));
+      socket.startHandshake();
+
+      return socket;
+    } catch (IOException e) {
+      instance.forceRefresh();
+      throw e;
+    }
+  }
+
+  DefaultConnectionInfoCache getConnection(ConnectionConfig config) {
+    return instances.computeIfAbsent(config, k -> createConnectionInfo(config));
+  }
+
+  private DefaultConnectionInfoCache createConnectionInfo(ConnectionConfig config) {
+    return new DefaultConnectionInfoCache(
+        config, adminApi, instanceCredentialFactory, executor, localKeyPair, minRefreshDelayMs);
+  }
+
+  public void forceRefresh() {
```

Comment:
This is unused and shouldn't be part the API.

Reply:
Removed. This isn't used.